### ISSUE TITLE
Do not use special characters in identifiers. `create extension "uuid-ossp";` (postgres)

### ISSUE THREAD

**Issue by NBprojekt** [sqlfluff/sqlfluff#4935]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When attempting to parse code to add extension with special characters in the name witch, witch need to be escaped, the linter fails with an error: `Do not use special characters in identifiers.`.

### Expected Behaviour

Should be able to add extensions with special characters in the name.

### Observed Behaviour

The check fails.

### How to reproduce

[Sqlfluff online](https://online.sqlfluff.com/fluffed?sql=H4sIAMV0nGQC_0suSk0sSVVIrShJzSvOzM9TUCotzUzRzS8uLlCy5uVKRpfOTFPIyy8BCmQWlxSjKQYAyU2fvkwAAAA%3D&dialect=postgres)

```
create extension "uuid-ossp";
create extension if not exists "uuid-ossp";
```

### Dialect

postgres

### Version

v2.1.1

### Configuration

[sqlfluff]
dialect = postgres
sql_file_exts = .sql

[sqlfluff:rules]
tab_space_size = 2
max_line_length = 160
indent_unit = space
comma_style = trailing

[sqlfluff:rules:L007]
operator_new_lines = before

[sqlfluff:rules:L057]
unquoted_identifiers_policy = all
quoted_identifiers_policy = all
allow_space_in_identifier = true
additional_allowed_characters = "-."

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by joaostorrer** [comment]
I tested this with version 2.3.5 (current) executing from cli with the configuration described here and it parses with no error.

```bash
$ sqlfluff parse --dialect postgres create_extension.sql
```

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_extension_statement:
[L:  1, P:  1]      |            keyword:                                          'create'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            keyword:                                          'extension'
[L:  1, P: 17]      |            whitespace:                                       ' '
[L:  1, P: 18]      |            extension_reference:
[L:  1, P: 18]      |                quoted_identifier:                            '"uuid-ossp"'
[L:  1, P: 29]      |    statement_terminator:                                     ';'
[L:  1, P: 30]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    statement:
[L:  2, P:  1]      |        create_extension_statement:
[L:  2, P:  1]      |            keyword:                                          'create'
[L:  2, P:  7]      |            whitespace:                                       ' '
[L:  2, P:  8]      |            keyword:                                          'extension'
[L:  2, P: 17]      |            whitespace:                                       ' '
[L:  2, P: 18]      |            keyword:                                          'if'
[L:  2, P: 20]      |            whitespace:                                       ' '
[L:  2, P: 21]      |            keyword:                                          'not'
[L:  2, P: 24]      |            whitespace:                                       ' '
[L:  2, P: 25]      |            keyword:                                          'exists'
[L:  2, P: 31]      |            whitespace:                                       ' '
[L:  2, P: 32]      |            extension_reference:
[L:  2, P: 32]      |                quoted_identifier:                            '"uuid-ossp"'
[L:  2, P: 43]      |    statement_terminator:                                     ';'
[L:  2, P: 44]      |    newline:                                                  '\n'
[L:  3, P:  1]      |    [META] end_of_file:
```

Perhaps is some custom configuration from site.

**Comment by R3gardless** [comment]
@greg-finley Are you still working on this issue? If not, would it be possible for me to take it over?

**Comment by alanmcruickshank** [comment]
I'm not working on it. Please go ahead! Welcome to the project 🎉

There have been a couple of recent PRs we've merged which might have some bearing on this (I don't think they've solved it, but they might help as templates). Check the change log to see which ones might be relevant.
